src/com/twu/biblioteca/MainMenu.java

```diff
@@ -1,40 +1,62 @@
 package com.twu.biblioteca;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Scanner;
 
 public class MainMenu {
 
-    public void welcome() {
-        System.out.println("Welcome to Biblioteca. Your one-stop-shop for great book titles in Bangalore!");
+    private final String welcome = "Welcome to Biblioteca. Your one-stop-shop for great book titles in Bangalore!";
+    private final String invalidMessage = "Please select a valid option!";
+    private final String goodBye = "Goodbye!";
+
+   ArrayList<String> books = new ArrayList<String>(Arrays.asList(
+           "1. Clean Code | Robert C. Martin | 2008",
+           "2. Don't Make Me Think | Steve Krug | 2000",
+           "3. Test Driven Development | Kent Beck | 2000"));
+
+   ArrayList<String> options = new ArrayList<String>(Arrays.asList(
+           "Please select an option from the following:",
+            "1. List of books",
+            "2. Quit"));
+
+    public void showWelcomeMessage() {
+        System.out.println(welcome);
+    }
+
+    public void showInvalidMessage() {
+        System.out.println(invalidMessage);
+    }
+
+    public void showGoodbyeMessage() {
+        System.out.println(goodBye);
     }
 
     public void showOptions() {
-        System.out.println("Please select an option from the following:");
-        System.out.println("1. List of books");
-        System.out.println("2. Quit");
+        for(int i = 0; i < options.size(); i++) {
+            System.out.println(options.get(i));
+        }
         System.out.print(">");
     }
 
     public void showList() {
-        System.out.println("1. Clean Code | Robert C. Martin | 2008\n" + "2. Don't Make Me Think | Steve Krug | 2000\n" + "3. Test Driven Development | Kent Beck | 2000\n");
-    }
-
-    public void showInvalidMessage() {
-        System.out.println("Please select a valid option!");
+        for(int i = 0; i < books.size(); i++) {
```

Comment:
This `for` loop will happily iterate over any number of things, but there is also a more specific way to iterate over an array which is slightly more concise as well... see if you can find it.

Reply:
Is it the for-each loop?

Reply:
Yep - unless you really _need_ the index (and normally you don't), the `for-each` loop is cleaner/simpler.